### PR TITLE
feat: Allow passing an identity for an address 

### PR DESCRIPTION
**Pull request by willemneal**
### What

Reuse the `identity::address` command to look up identities when parsing an `Address`.

### Why

Allows aliases instead of having to pass the public key.

### Known limitations

[TODO or N/A]

### CODE REVIEW

File: .cargo/config.toml
Comment (paulbellamy) [inline .cargo/config.toml:1]: Accidental commit?

File: .github/workflows/full-help-docs.yml
Comment (paulbellamy) [inline .github/workflows/full-help-docs.yml:1]: Also seems unrelated?

File: cmd/soroban-cli/src/commands/contract/invoke.rs [hunk 1]
```diff
@@ -189,7 +189,17 @@ impl Cmd {
             .map(|i| {
                 let name = i.name.to_string().unwrap();
                 if let Some(mut raw_val) = matches_.get_raw(&name) {
-                    let s = raw_val.next().unwrap().to_string_lossy().to_string();
+                    let mut s = raw_val.next().unwrap().to_string_lossy().to_string();
+                    if matches!(i.type_, ScSpecTypeDef::Address) {
+                        let cmd = crate::commands::config::identity::address::Cmd {
+                            name: Some(s.clone()),
+                            hd_path: Some(0),
```
Comment (paulbellamy) [inline cmd/soroban-cli/src/commands/contract/invoke.rs:196]: Unrelated side note: Been thinking about this hd-path thing more, and it feels like a sort of leaky abstraction in identities. What if it was rolled into a named identity thing? So, instead of passing `--hd-path` on the CLI you'd first have to do:
```
# Explicitly generate a derived subpath key for usage, something like
$ soroban config identity generate --parent existing_identity --hd-path 1 child_name

# and then you'd use it like a normal identity
$ soroban contract invoke --source child_name ...
```
Thoughts?

File: cmd/soroban-cli/src/strval.rs [hunk 2]
```diff
@@ -82,14 +82,16 @@ impl Spec {
...
-            | ScType::Address
             | ScType::Void
             | ScType::Timepoint
             | ScType::Duration
             | ScType::U256
             | ScType::I256
             | ScType::String
             | ScType::Bool => String::new(),
+            ScType::Address => String::from(
+                "Can be public key (G13..), a contract hash (6c45307) or an identity (alice), ",
```
Comment (paulbellamy) [inline cmd/soroban-cli/src/strval.rs:93]: Or an `S...` key?

Comment (paulbellamy) [reply cmd/soroban-cli/src/strval.rs:93]: Also, we should get rid of contract hash support here when we do https://github.com/stellar/soroban-tools/issues/403

Comment (willemneal) [reply cmd/soroban-cli/src/strval.rs:93]: Yeah we could, but it's a bad practice to put a secret in plaintext in a CLI command.   But for scripts I guess it's okay.

File: .github/workflows/full-help-docs.yml
Comment (willemneal) [reply .github/workflows/full-help-docs.yml:1]: Yeah.  I'll remove these files.

File: cmd/soroban-cli/src/commands/contract/invoke.rs (hunk 1 above)
Comment (willemneal) [reply cmd/soroban-cli/src/commands/contract/invoke.rs:196]: I was thinking that we could also do something like `existing_id:<hd_path>` then that id file can just reference the other id.

File: .cargo/config.toml
Comment (willemneal) [reply .cargo/config.toml:1]: Ah yeah this was meant for another PR. But it ensures that the clippy warnings happen when building and adds an alias for markdown

**Review by paulbellamy — APPROVED** [review]
